src/test/java/org/sonatype/nexus/ci/nxrm/v3/CreateTagBuilderTest.groovy

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.nxrm.v3
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+
+import com.sonatype.nexus.api.exception.RepositoryManagerException
+import com.sonatype.nexus.api.repository.v3.RepositoryManagerV3Client
+import com.sonatype.nexus.api.repository.v3.Tag
+
+import org.sonatype.nexus.ci.config.GlobalNexusConfiguration
+import org.sonatype.nexus.ci.config.Nxrm2Configuration
+import org.sonatype.nexus.ci.config.Nxrm3Configuration
+import org.sonatype.nexus.ci.nxrm.Messages
+import org.sonatype.nexus.ci.nxrm.v3.CreateTagBuilder.DescriptorImpl
+import org.sonatype.nexus.ci.util.RepositoryManagerClientUtil
+
+import hudson.model.Result
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition
+import org.jenkinsci.plugins.workflow.job.WorkflowJob
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import org.jvnet.hudson.test.JenkinsRule
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static hudson.util.FormValidation.Kind.ERROR
+import static hudson.util.FormValidation.Kind.OK
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_Validation_TagAttributesJson
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_Validation_TagNameEmpty
+import static org.sonatype.nexus.ci.nxrm.Messages.CreateTag_Validation_TagNameInvalid
+
+class CreateTagBuilderTest
+    extends Specification
+{
```

Comment:
Not sure if or how much value it adds a value but I included a test using the generated workflow dsl which allows you to verify behavior if incorrect inputs are provided as a result of a user error in entering.  I could go either way on it.